app/src/org/commcare/views/EntityDetailView.java

```diff
@@ -320,6 +330,21 @@ public boolean onTouch(View view, MotionEvent event) {
 
             graphLayout.removeAllViews();
             graphLayout.addView(graphView, GraphView.getLayoutParams());
+            if (showSpinner) {
+                final ProgressBar spinner = new ProgressBar(this.getContext(), null, android.R.attr.progressBarStyleLarge);
+                spinner.setLayoutParams(new FrameLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT, Gravity.CENTER));
+                GraphLoader graphLoader = new GraphLoader((Activity) this.getContext(), spinner);

```

Comment:
I _don't think_ this is an issue given that we are providing all the JS being run here...